### PR TITLE
doc: nix-gitignore to CommonMark

### LINKED ISSUE #125670: Convert nix-gitignore from Docbook to CommonMark
This issue tracks the conversion of `/doc/functions/nix-gitignore.xml` from Docbook to CommonMark per [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).

If you intend to work on this conversion, please comment below!

# Conversion instructions

1. Comment here that you intend to work on it. (Please also post if you stop working on it.)
2. Checkout a new branch of nixpkgs master.
3. Edit `/doc/functions.xml`: change `nix-gitignore.xml` to `nix-gitignore.section.xml`.
4. Rename `/doc/functions/nix-gitignore.xml` to `nix-gitignore.section.md`.
5. Convert the contents from Docbook to [CommonMark](https://commonmark.org/). Follow [conversion best practices](https://github.com/NixOS/nixpkgs/issues/105243). Use existing `.md` sections for inspiration.
6. Use `nix-build` from the `doc/` folder to build the manual.
7. Preview the manual on your computer at `http://0.0.0.0:8000/manual.html` after running from `doc/` folder:
  ```
  nix-shell -p python3 --run 'python3 -m http.server --directory ./result/share/doc/nixpkgs'
  ```
8. When it looks good, submit a PR and reference this issue.

### PR DESCRIPTION
**Pull request by alarsyo**
###### Motivation for this change

Closes #125670

Rendered: 
image

### CODE REVIEW

File: doc/functions/nix-gitignore.section.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+These functions are derived from the `Filter` functions by setting the first filter argument to `(_: _: true)`:
+
+```nix
+gitignoreSourcePure = gitignoreFilterSourcePure (_: _: true);
+gitignoreSource = gitignoreFilterSource (_: _: true);
+```
+
+Those filter functions accept the same arguments the `builtins.filterSource` function would pass to its filters, thus `fn: gitignoreFilterSourcePure fn ""` should be extensionally equivalent to `filterSource`. The file is blacklisted iff it\'s blacklisted by either your filter or the gitignoreFilter.
```
Comment (rissson) [inline doc/functions/nix-gitignore.section.md:33]:
```suggestion
Those filter functions accept the same arguments the `builtins.filterSource` function would pass to its filters, thus `fn: gitignoreFilterSourcePure fn ""` should be extensionally equivalent to `filterSource`. The file is blacklisted if it\'s blacklisted by either your filter or the gitignoreFilter.
```

Comment (alarsyo) [reply doc/functions/nix-gitignore.section.md:33]: I figured that was the contraction of https://en.m.wikipedia.org/wiki/If_and_only_if ?

Comment (alarsyo) [reply doc/functions/nix-gitignore.section.md:33]: Not sure that make sense in that case though, indeed

Comment (rissson) [reply doc/functions/nix-gitignore.section.md]: If it is indeed `iff`, I guess we can expand it for people like me won't don't understand those.

**Review by ryantm — APPROVED** [review]

**Pull request merged**